collab2/foraging/toolkit/proximity.py

```diff
@@ -1,93 +1,209 @@
+                                          function.
+    :param proximity_function: A callable function used to compute proximity scores based on distance.
+                               Defaults to `_piecewise_proximity_function`.
+    :param proximity_function_kwargs: Additional keyword arguments for the proximity function.
+
+    :return: A list of lists of pandas DataFrames, where each DataFrame has been updated with the computed
+             proximity predictor values.
+    """
+
+    num_foragers = len(foragers)
+    num_frames = len(foragers[0])
+    predictor = copy.deepcopy(local_windows)
+
+    for f in range(num_foragers):
+        for t in range(num_frames):
+            if predictor[f][t] is not None:
+                # add column for predictor_name
+                predictor[f][t][predictor_name] = 0
+                # find confocals within interaction length
+                interaction_partners = filter_by_distance(
+                    foragersDF,
+                    f,
+                    t,
+                    interaction_length,
+                    interaction_constraint,
+                    interaction_constraint_params,
+                )
 
-    forager_distances = foragers_to_forager_distances(foragers)
+                if len(interaction_partners) > 0:
+                    for partner in interaction_partners:
+                        partner_x = foragers[partner]["x"].iloc[t]
+                        partner_y = foragers[partner]["y"].iloc[t]
+
+                        predictor[f][t][
+                            predictor_name
+                        ] += _proximity_predictor_contribution(
+                            partner_x,
+                            partner_y,
+                            local_windows[f][t],
+                            proximity_function,
+                            **proximity_function_kwargs,
+                        )
 
-    proximity = visibility.copy()
-    for b in range(len(foragers)):
-        for t in range(start, end):
-            proximity[b][t]["proximity"] = 0
+                # scaling to abs max (not sum, as this would lead to small numerical values)
+                max_abs_over_grid = predictor[f][t][predictor_name].abs().max()  # sum()
+                if max_abs_over_grid > 0:
+                    predictor[f][t][predictor_name] = (
+                        predictor[f][t][predictor_name] / max_abs_over_grid
+                    )
 
-            distbt = forager_distances[b].iloc[t].drop(b + 1)
+    return predictor
 
-            visible_foragers = distbt[distbt <= visibility_range].index.tolist()
 
-            if len(visible_foragers) > 0:
-                for vb in range(len(visible_foragers)):
-                    o_x = foragers[visible_foragers[vb] - 1]["x"].iloc[t]
-                    o_y = foragers[visible_foragers[vb] - 1]["y"].iloc[t]
+def generate_proximity(foragers_object: dataObject, predictor_name: str):
```

Comment:
pls rename to `generate_proximity_predictor`

Reply:
done